azurefunctions-extensions-bindings-cosmos/azurefunctions/extensions/bindings/cosmos/cosmosClient.py

```
@@ -0,0 +1,57 @@
+
+
+class CosmosClient(SdkType):
+    def __init__(self, *, data: Datum) -> None:
+        # model_binding_data properties
+        self._data = data
+        self._version = None
+        self._source = None
+        self._content_type = None
+        self._database_name = None
+        self._container_name = None
+        self._connection = None
+        self._using_managed_identity = False
+        self._partition_key = None
+        self._sql_query = None
+        self._preferred_locations = None
+        if self._data:
+            self._version = data.version
+            self._source = data.source
+            self._content_type = data.content_type
+            content_json = json.loads(data.content)
+            self._database_name = content_json.get("DatabaseName")
+            self._container_name = content_json.get("ContainerName")
+            self._connection = get_connection_string(content_json.get("Connection"))
+            self._using_managed_identity = using_managed_identity(
+                content_json.get("Connection")
+            )
+            self._preferred_locations = content_json.get("PreferredLocations")
+
+    def get_sdk_type(self) -> CosmosClientSdk:
+        """
+        There are two ways to create a CosmosClient:
+        1. Through the constructor: this is the only option when using Managed Identity
+        2. Through from_connection_string: this is the only option when not using Managed Identity
+
+        We track if Managed Identity is being used through a flag.
+        """
+        if self._data:
+            cosmos_client = (
+                CosmosClientSdk(
+                    url=self._connection, credential=DefaultAzureCredential()
+                )
+                if self._using_managed_identity
+                else CosmosClientSdk.from_connection_string(self._connection)
+            )
+            return cosmos_client
+        else:
+            return None
```

Thinking on this, should we raise an exception if the client cannot be created?

---

Testing this:
If we return None
Exception: AttributeError: 'NoneType' object has no attribute 'list_databases' - so any operation with the client will throw since it's None

If we raise the exception then the ux will get error message directly and immediately - this is better imo

This should also be done in all the other bindings too